src/main/java/frc/robot/subsystems/vision/SImVisionIO.java

```diff
@@ -35,29 +38,35 @@ private SImVisionIO(int camIndex) {
         }
     }
 
-
     @Override
     public void setPipeLine(int pipeLineIndex) {
         simCamera.getCamera().setPipelineIndex(pipeLineIndex);
     }
 
     @Override
     public Result getLatestResult() {
-        return new Result(simCamera.getCamera().getLatestResult().getTimestampSeconds(), new Pose3d(simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getTranslation(),simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getRotation()));
+        return new Result(simCamera.getCamera().getLatestResult().getTimestampSeconds(), new Pose3d(simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getTranslation(), simCamera.getCamera().getLatestResult().getBestTarget().getBestCameraToTarget().getRotation()));
     }
 
     @Override
-    public void updateInputs(VisionInputs inputs) {
+    public void updateInputs(VisionInputsAutoLogged inputs) {
 
         for (int i = 1; i < VisionConstants.TARGET_POSITION_SIM.length; i++) {
             coprocessorSim.addVisionTargets(VisionConstants.SIM_VISION_TARGETS[i]);
             coprocessorSim.update(SwerveDrive.getInstance().getBotPose());
+            estimatedPose = estimator.update(simCamera.getCamera().getLatestResult()).get().estimatedPose;
         }
-        estimator.update();
-
 
+        inputs.hasTargets = simCamera.getCamera().getLatestResult().hasTargets();
+        inputs.yaw = simCamera.getCamera().getLatestResult().getBestTarget().getYaw();
+        inputs.pitch = simCamera.getCamera().getLatestResult().getBestTarget().getPitch();
+        inputs.area = simCamera.getCamera().getLatestResult().getBestTarget().getArea();
+        inputs.targetSkew = simCamera.getCamera().getLatestResult().getBestTarget().getSkew();
+        inputs.targetID = simCamera.getCamera().getLatestResult().getBestTarget().getFiducialId();
+        inputs.cameraToTarget = new double[]{bestCameraToTarget.getX(), bestCameraToTarget.getY(), bestCameraToTarget.getZ(), bestCameraToTarget.getRotation().getX(), bestCameraToTarget.getRotation().getY(), bestCameraToTarget.getRotation().getZ()};
+        inputs.poseFieldOriented = new double[]{estimatedPose.getX(), estimatedPose.getY(), estimatedPose.getZ(), estimatedPose.getRotation().getQuaternion().getW(), estimatedPose.getRotation().getQuaternion().getX(), estimatedPose.getRotation().getQuaternion().getY(), estimatedPose.getRotation().getQuaternion().getZ()};
+        inputs.fieldOrientedRotationRad = new double[]{estimatedPose.getRotation().getX(), estimatedPose.getRotation().getY(), estimatedPose.getRotation().getZ()};
```

Comment:
`Pose3d` and `Transform3d` have struct support, so you can just log them directly without needing to convert them to an array.